javascript-practice/src/js/views/product.view.js

```diff
@@ -0,0 +1,69 @@
+import getColorOptionList from "./getColorOptionList";
+
+export default class ProductView {
+  /**
+   * Displays product list of products on the view
+   * @param {Object[]} products
+   */
+  displayProducts(products) {
+    // Mapping over the products array to create HTML elements for each product
+    const productElements = products.map(product => {
+      // Creating the main div element for each product
+      const productItemElement = document.createElement('div');
+      productItemElement.className = 'product-item';
+
+      // Creating the figure element for the product image
+      const productImageFigureElement = document.createElement('figure');
+      productImageFigureElement.className = 'product-thumbnail';
+
+      const productLinkElement = document.createElement('a');
+
+      productLinkElement.href = '';
+
+      // Creating the img element for the product image
+      const productImageElement = document.createElement('img');
+      productImageElement.src = product.imgUrl;
+      productImageElement.alt = product.name;
+
+      productLinkElement.append(productImageElement);
+      productImageFigureElement.append(productLinkElement);
+
+      // Creating the div element for the product details
+      const productDetailElement = document.createElement('div');
+
+      // Creating the h2 element for the product name
+      const productNameElement = document.createElement('h2');
+      productNameElement.className = 'product-info';
+      productNameElement.textContent = product.name;
+
+      // Creating the ul element for the product colors
+      const productColorsWrapperElement = document.createElement('ul');
+      const colorOptionList = getColorOptionList(product.colors);
+      productColorsWrapperElement.append(...colorOptionList);
+
+      // Creating the p element for the product price
+      const productPriceElement = document.createElement('p');
```

Comment:
To avoid duplicate code, you can create a common function same as below 

```
function createElement(tag, className, textContent = '') {
    const element = document.createElement(tag);
    element.className = className || '';
    element.textContent = textContent;
    return element;
}```